admin/scripts/modules/aktivity/prihlaseni-drd.php

```diff
@@ -11,8 +11,21 @@
 if (post('vypadliSemifinale') || post('vypadliFinale')) {
     $a = Aktivita::zId(post('zakladni'));
     $aVypadli = post('vypadliSemifinale') ? Aktivita::zId(post('semifinale')) : Aktivita::zId(post('finale'));
+
+    $mail = new GcMail();
+    $mail->predmet('Gamecon: umístění družiny v MDrD');
+    $druzina = $a->tym()->nazev();
+
     foreach ($a->prihlaseni() as $uc) {
         $aVypadli->odhlas($uc, Aktivita::BEZ_POKUT);
+
+        $mail->text("Ahoj " . $uc->nick() . ",\n
+            bohužel, tvoje družina " . $druzina . " nepostoupila do dalšího kola. Herní bloky, původně 
```

Comment:
Vypadá to v pohodě. Chtěl bych se ale mrknout do zadání - je na to nějaká karta v Trellu?

Reply:
Yes, tady: https://trello.com/c/q1Gdu1lG (příště přihodím rovnou).
Myslíš, že je kdyžtak nějaký dobrý způsob, jak to předat Guffovi k otestování (krom ostrého provozu :D)? Nezmiňoval jsi něco jako demo.gamecon.cz? Promiň, že mám teď tolik nejasností, jak se do toho dostanu, tak to snad bude lepší :)

Reply:
My jsme postupem času zjistili, že je dobré to dávat rovnou do textu k pull requestu.

Díky, vypadá to dobře.

Snad to jen nebude trvat věky, než se to celé synchronně provede (na asynchronní maily nemáme nástroje - to bychom museli řešit přes nějaký MailChimp nebo tak). PHP proces má standardní timeout je 30 sekund https://www.php.net/manual/en/function.set-time-limit.php, to by měl stihnout poslat dvě stě mailů (nebo kolik to bude).

Reply:
Myslel jsem na ten timeout, ale tohle by mělo fungovat tak, že se to odešle jen dané družině, tj. kolem 5 mailů. Takže to by neměl být problém.

Reply:
Já pořád žiju v tom, že když to bude aktivita ve smyslu "První den mDrD", tak na ní bude přihlášeno několik set lidí.

Ale vůbec si s tím nejsem jist a budu se na to muset zeptat Guffa.

Reply:
Tento event se spustí na tlačítko u teamu, kde je napsaný něco jako "vypadli ve čtvrtfinále" nebo "vypadli v semifinále". Procházel jsem ten kód a mělo by to být OK, ale proto to bude lepší otestovat, no. Cemi říkala, že až bude mít chvilku, naplní betu pár aktivitama a zkusí to.